### ISSUE TITLE
feat: add optgroup to select option

### ISSUE THREAD

**Issue by JulioCVaz** [ionic-team/ionic-framework#28628]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

I think is interesting to include ```html <optgroup> ``` tag for ion-select. If is possible I wanna to implement this.

References: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup

### Describe the Use Case

Use ion-select with optgroup tag, like:

```js
<IonSelect
          aria-label="Fruit"
          placeholder="Select fruit"
          onIonChange={(e) => console.log(`ionChange fired with value: ${e.detail.value}`)}
          onIonCancel={() => console.log('ionCancel fired')}
          onIonDismiss={() => console.log('ionDismiss fired')}
>
         <IonOptGroup label="Fruits">
             <IonSelectOption value="apples">Apples</IonSelectOption>
             <IonSelectOption value="oranges">Oranges</IonSelectOption>
             <IonSelectOption value="bananas">Bananas</IonSelectOption>
          </IonOptGroup>
</IonSelect>
```

**Comment by JulioCVaz** [comment]
> Thanks for the report. Can you help me understand what you are trying to accomplish? In the example you provided you can specify a group header using either the `label` property on `ion-select` or `interfaceOptions`: https://ionicframework.com/docs/api/select#interface-options

Sure @liamdebeasi

The ideia is to create a new tag to support `optgroup` in [selects](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup), like in HTML5.

The **interfaceOptions** is awesome to create headers and subheaders, but is possible to use in only one Select, correct?

Simplifying, is to add groups to ion-select :)

**Comment by liamdebeasi** [comment]
> The interfaceOptions is awesome to create headers and subheaders, but is possible to use in only one Select, correct?

That's correct. The reason why we only have one group in an `ion-select` is because alert and action sheet only support single groups.

What problem are you trying to solve by adding multiple groups? Additionally, what are some of the alternatives you have tried?

**Comment by JulioCVaz** [comment]
> > The interfaceOptions is awesome to create headers and subheaders, but is possible to use in only one Select, correct?
> 
> That's correct. The reason why we only have one group in an `ion-select` is because alert and action sheet only support single groups.
> 
> What problem are you trying to solve by adding multiple groups? Additionally, what are some of the alternatives you have tried?

Perfect @liamdebeasi . In our company we have a some of selects that using [optgroup](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup) to select stores and store items for exemple. Like: 

Store 1
   t-shirt
Store 2
  Shoes

And recently we created a new [design system](https://github.com/juntossomosmais/atomium) for our UI. But we has this problem to use optgroups. To avoid this, we use disabled options as labels and normal options in the select.

**Comment by liamdebeasi** [comment]
Which interface are you using? (alert, action sheet, popover)

**Comment by JulioCVaz** [comment]
> Which interface are you using with `ion-select`? (alert, action sheet, popover)

is popover interface 👍

**Comment by liamdebeasi** [comment]
Thanks for the additional info. We had [a similar request](https://github.com/ionic-team/ionic-framework/issues/16809) a few years ago, so we'd like to keep this open and collect community feedback before proceeding.

**Comment by btsiders** [comment]
I love this idea. It would help in cases where I have enough options in the select to need a little bit of grouping, but not enough options to split the select into one for groups and one for that group's options.

**Comment by JulioCVaz** [comment]
Hey @liamdebeasi, I hope you are well!

any update for the development of this comp? Can I help or develop it? :)

**Comment by josuelmm** [comment]
Will they never integrate this? It's a basic thing about selects.

**Comment by sembo199** [comment]
I would like to see this feature in Ionic as well. Especially when using long lists of select options, I would want to be able to do some extra grouping for a better user experience. Additionally, I think a widely supperted HTML5 element should for sure be useable in Ionic.
